src/utils/generateUID.js

```diff
@@ -0,0 +1,22 @@
+export const generateUID = (prefix, separator) => {
+  let counter = 1
+
+  let map = new WeakMap()
+
+  const get = item => {
+    if (!map.has(item)) {
+      map.set(item, counter++)
+      return get(item)
```

Comment:
I guess we can remove the extra function call here? Without the return, it'll execute the same fall through logic if I'm not mistaken.